### PR TITLE
Remove support for build to cache for codegen

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

To make codegen compatible with flutter web, we need to enforce build to source instead of cache. This allows two separate build_runner processes (user supplied and flutter tool) to operate on the same application without needing to coordinate on what the correct multiroot scheme is.

### CODE REVIEW

File: packages/flutter_test/lib/src/window.dart [hunk 1]
```diff
@@ -347,6 +347,11 @@ class TestWindow implements Window {
     _window.onPlatformMessage = callback;
   }
 
+  @override
+  dynamic noSuchMethod(Invocation invocation) {
```
Comment (yjbanov) [inline packages/flutter_test/lib/src/window.dart]: What does this change do?

**Review by yjbanov — COMMENTED** [review]
But generating to source is bad. No?

File: packages/flutter_test/lib/src/window.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_test/lib/src/window.dart]: Oh my bad!  I was testing some things, there are still some incompatibilities with the web ui

**Comment by jonahwilliams** [comment]
It is, but the most popular builders are already using it, and build to cache requires us to track the multiroot filesystem and coordinate with the web rules in ways that aren't currently supported

**Comment by yjbanov** [comment]
Sorry, I have to probe deeper, as this could have long-term repercussions.

> the most popular builders are already using it

Do you think they are using it because they want to or because they have to? :thinking: 

> build to cache requires us to track the multiroot filesystem and coordinate with the web rules in ways that aren't currently supported

Is it not supported in `build_runner` or in `flutter_tool`? What would it take to fix it?

Finally, how does this affect our ability to run builders on dependent packages?

**Comment by jonahwilliams** [comment]
Lets chat about this some more when i get back. There are other approaches here

**Pull request closed without merging**